### DISCUSSION TITLE
signal_jni error when running locally built signal-cli

### DISCUSSION THREAD

**Discussion by JpGallegos** [AsamK/signal-cli#476] [Q&A] [answered]
Hello,

### Issue
When running `signal-cli` after building it I get the following output:
```sh
gradle run --args="-u +1<my phone number> register"

> Task :run FAILED
WARN App - WARNING: Support for new group V2 is disabled, because the required native library dependency is missing: libzkgroup
ERROR App - Error loading or creating state file: no signal_jni in java.library.path: /Users/jgallegos/Library/Java/Extensions:/Library/Java/Extensions:/Network/Library/Java/Extensions:/System/Library/Java/Extensions:/usr/lib/java:.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command '/usr/local/Cellar/openjdk/15.0.1/libexec/openjdk.jdk/Contents/Home/bin/java'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
5 actionable tasks: 1 executed, 4 up-to-date
```

### Environment:
  OS: Mac OS X 11.1 x86_64
  Gradle: 6.8.1
  JDK: "11.0.10" 2021-01-19 LTS
  Commit: a811d1a05a28d332705f30528db882a1031e8257

**Comment by finnertysea** [comment]
I get the same signal_jni error! I hope this is solved soon

I am running Mac OS X 10.15.7, Gradle 6.8.1, and JDK 15.0.2

**Comment by AsamK** [comment, marked as answer]
See https://github.com/AsamK/signal-cli/wiki/Provide-native-lib-for-libsignal

**Comment by JpGallegos** [comment]
Thanks to the guide @AsamK linked I found that the issue is that OSX doesn't like the `.so` objects in the `.jar` files. To get it to run correctly you need to put `libsignal.dylib` and `libzkgroup.dylib` somewhere in your java library path. In my case I put them in `~/Library/Java/Extensions`

**Reply by zmijunkie** [reply]
libzkgroup.dylib was downloadable at https://github.com/signalapp/zkgroup/releases/download - but where did you get libsignal.dylib from ?
Building libsignal-client did not help me ...

**Reply by JpGallegos** [reply]
I don't know what to tell you, I followed the steps [in the wiki page](https://github.com/AsamK/signal-cli/wiki/Provide-native-lib-for-libsignal#libsignal-client) and got the `libsignal.dylib` exactly where the wiki said it would be.


<img width="1248" alt="Screen Shot 2021-02-13 at 11 19 38 AM" src="https://user-images.githubusercontent.com/4839833/107853608-60953500-6ded-11eb-85d1-0952798181d4.png">
